From c7062cc453c8e98d4cd532acfd2169afab2869aa Mon Sep 17 00:00:00 2001
From: aottaviano <aottaviano@iis.ee.ethz.ch>
Date: Sat, 1 Apr 2023 21:07:52 +0200
Subject: [PATCH 3/7] platform: Update to cheshire's new memory layout

---
 platform/fpga/cheshire/platform.c | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/platform/fpga/cheshire/platform.c b/platform/fpga/cheshire/platform.c
index 407489f..9da58c1 100644
--- a/platform/fpga/cheshire/platform.c
+++ b/platform/fpga/cheshire/platform.c
@@ -18,19 +18,19 @@
 #include <sbi_utils/serial/uart8250.h>
 #include <sbi_utils/timer/aclint_mtimer.h>
 
-#define CHESHIRE_UART_ADDR			        0x02003000
-#define CHESHIRE_UART_FREQ			        50000000
-#define CHESHIRE_UART_BAUDRATE			    115200
-#define CHESHIRE_UART_REG_SHIFT			    2
-#define CHESHIRE_UART_REG_WIDTH			    4
-#define CHESHIRE_PLIC_ADDR			        0x0c000000
-#define CHESHIRE_PLIC_NUM_SOURCES			20
-#define CHESHIRE_HART_COUNT			        1
-#define CHESHIRE_CLINT_ADDR			        0x04000000
-#define CHESHIRE_ACLINT_MTIMER_FREQ		    1000000
-#define CHESHIRE_ACLINT_MSWI_ADDR			(CHESHIRE_CLINT_ADDR + 0x0)
-#define CHESHIRE_ACLINT_MTIMER_ADDR		    (CHESHIRE_CLINT_ADDR + 0xbff8)
-#define CHESHIRE_ACLINT_MTIMECMP_ADDR       (CHESHIRE_CLINT_ADDR + 0x4000)
+#define CHESHIRE_UART_ADDR	      0x03002000
+#define CHESHIRE_UART_FREQ	      50000000
+#define CHESHIRE_UART_BAUDRATE	      115200
+#define CHESHIRE_UART_REG_SHIFT	      2
+#define CHESHIRE_UART_REG_WIDTH	      4
+#define CHESHIRE_PLIC_ADDR	      0x04000000
+#define CHESHIRE_PLIC_NUM_SOURCES     20
+#define CHESHIRE_HART_COUNT	      1
+#define CHESHIRE_CLINT_ADDR	      0x02040000
+#define CHESHIRE_ACLINT_MTIMER_FREQ   1000000
+#define CHESHIRE_ACLINT_MSWI_ADDR     (CHESHIRE_CLINT_ADDR + 0x0)
+#define CHESHIRE_ACLINT_MTIMER_ADDR   (CHESHIRE_CLINT_ADDR + 0xbff8)
+#define CHESHIRE_ACLINT_MTIMECMP_ADDR (CHESHIRE_CLINT_ADDR + 0x4000)
 
 
 static struct plic_data plic = {
-- 
2.31.1

