diff --git a/package/ccache/ccache.mk b/package/ccache/ccache.mk
index ac4bc4aba5..32829dfbe6 100644
--- a/ccache.mk
+++ b/ccache.mk
@@ -16,7 +16,9 @@ HOST_CCACHE_CONF_OPTS += \
 	-UCMAKE_C_COMPILER_LAUNCHER \
 	-UCMAKE_CXX_COMPILER_LAUNCHER \
 	-DZSTD_FROM_INTERNET=OFF \
-	-DHIREDIS_FROM_INTERNET=OFF
+	-DHIREDIS_FROM_INTERNET=OFF \
+	-DHAVE_C_AVX512=no
+
 
 # Patch host-ccache as follows:
 #  - Use BR_CACHE_DIR instead of CCACHE_DIR, because CCACHE_DIR
